tests/network/curl_client.pm

```diff
@@ -0,0 +1,42 @@
+# SUSE"s openQA tests
+# #
+# # Copyright © 2019 SUSE LLC
+# #
+# # Copying and distribution of this file, with or without modification,
+# # are permitted in any medium without royalty provided the copyright
+# # notice and this notice are preserved.  This file is offered as-is,
+# # without any warranty.
+#
+# Summary: Test regression to curl.
+#  this test curl with http, https, ldap, ftp and ntlm auth
+#  poo#51536
+# Maintainer: Marcelo Martins <mmartins@suse.cz>
+
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
+use lockapi qw(mutex_create mutex_wait);
+
+sub run {
+    my $self = shift;
+    $self->select_serial_terminal;
+
+    #waiting server ready.
+    mutex_wait('curl_server_ready', undef, 'Server configuration in progress!');
+    record_info 'Waiting Server', 'Waiting Curl-server to start tests.';
+
+    #start curl tests
+    assert_script_run('curl -f -v http://10.0.2.101/get 2>&1');
+    assert_script_run('curl -f -v https://httpbin.org/get 2>&1');
+    assert_script_run('curl "ldap://10.0.2.101/dc=example,dc=com??sub?(uid=bjensen)"');
+    assert_script_run('curl ftp://10.0.2.101');
+
+    assert_script_run('curl -u "COLLECTIVE\Administrator":"password" --ntlm http://borg.qam.suse.de/test/undercover_test.txt');
+
+    #finish test
+    mutex_create('curl_client_ready');
```

Comment:
Why you have two mutexes (curl_client_ready & curl_test_sessions) here ? Nothing is happening between this two mutexes. One like 'curl_client_done' should be enough.

Reply:
And as server is `wait_for_children;` you don't need any mutex at all ?

Reply:
Ok, I think solved now. I ran tests again. Thanks.